### PR TITLE
ui/shared/profile/IcicleGraphArrow: Create arrow-based IcicleGraph

### PR DESCRIPTION
**Pull request by metalmatze**
These are the frontend changes to read flame graphs from arrow records. 
The backend changes are in https://github.com/parca-dev/parca/pull/3069 which this PR depends on.

### CODE REVIEW

**Review by manojVivek — APPROVED** [review]
Lgtm overall!

File: ui/packages/shared/profile/src/ProfileIcicleGraph/IcicleGraphArrow/index.tsx [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    return <></>;
+  }
+
+  return (
+    <div onMouseLeave={() => dispatch(setHoveringNode(undefined))}>
+      <select
+        className="rounded-md border bg-gray-50 px-3 py-2 text-sm shadow-sm focus:border-indigo-500 focus:outline-none focus:ring-1 focus:ring-indigo-500 dark:bg-gray-900"
+        onChange={e => setSortBy(e.target.value)}
+      >
+        <option value="function_name">Function Name</option>
```
Comment (yomete) [inline ui/packages/shared/profile/src/ProfileIcicleGraph/IcicleGraphArrow/index.tsx:168]: Shall we place the select dropdown next to the `Reset View` button and add a text in front. Like this:

"Reset View"  "Sort by: `<Select />`"

File: ui/packages/shared/profile/src/ProfileView/index.tsx [hunk 2]
```diff
@@ -52,7 +53,8 @@ type NavigateFunction = (path: string, queryParams: any, options?: {replace?: bo
 
 export interface FlamegraphData {
   loading: boolean;
-  data?: Flamegraph;
```
Comment (yomete) [inline ui/packages/shared/profile/src/ProfileView/index.tsx:55]: We might need to cater for this removed prop for the grafana plugin as that's the reason for the typing error [here](https://github.com/parca-dev/parca/actions/runs/5457111282/jobs/9930779290?pr=3326).

**Review by yomete — APPROVED** [review]
lgtm overall! just a few things to work over that i mentioned.

File: ui/packages/shared/profile/src/ProfileView/index.tsx (hunk 2 above)
Comment (metalmatze) [reply ui/packages/shared/profile/src/ProfileView/index.tsx:55]: aah, I see. Let's not rename it then, but only add the table one for now. I think there is a bigger refactoring we should do anyway.

**Comment by metalmatze** [comment]
There are some longer-running CI checks still running, but I would like to get this in before we have another conflict. Happy to follow up with PRs if anything is found.

**Pull request merged**